### DISCUSSION TITLE
Questions about the license and allowed usage

### DISCUSSION THREAD

**Discussion by reemrizzk** [cefsharp/CefSharp#4350] [Q&A]
Hello, I have questions about the license of CefSharp, I read the license but still don't understand it very well. This is how I want to use CefSharp: 

1- **is the below usage allowed?**
 I am developing a web design software which is a c# WinForms application and I want to use cefsharp in my WinForms project to display the page output. (I am currently using WebBrowser in my app for that but it is too old and I want to use a more modern browser in my app). My WinForms app is currently free to download but its **not open source**, which means I didn't share the app's code. Also, although the app is free to download, some of the plugins I am offering for the app, are **not for free**. the plugins themselves don't have webview browser in their code, but might call functions in the main app that updates the browser. 

2- I read on the website "CefSharp is [BSD licensed](https://opensource.org/licenses/BSD-3-Clause), which means that it can be used for proprietary and free/open source applications. If you are using CefSharp in a commercial application please consider sponsoring the maintainer via [GitHub Sponsors](https://github.com/sponsors/amaitland)." 
**is sponsoring required/mandatory if I want to use cefsharp in a commercial app or it is only preferred/appreciated?**

thanks

**Comment by amaitland** [comment]
I'm not a lawyer, nothing I say should be considered legal advice.

> CefSharp is [BSD licensed](https://opensource.org/licenses/BSD-3-Clause), which means that it can be used for proprietary and free/open source applications.

If you need further clarification on `BSD 3-Clause License` then I'd suggest reading https://fossa.com/blog/open-source-software-licenses-101-bsd-3-clause-license/ 

Long as you comply with the `License` then I think you should be fine.

- [CefSharp License](https://github.com/cefsharp/CefSharp/blob/master/LICENSE)
- [CEF License](https://github.com/chromiumembedded/cef/blob/master/LICENSE.txt)
- [Chromium License](https://chromium.googlesource.com/chromium/src/+/HEAD/LICENSE)
- Load `chrome://credits` in a `ChromiumWebBrowser` instance to see further `Chromium` dependency licenses.

> is sponsoring required/mandatory if I want to use cefsharp in a commercial app or it is only preferred/appreciated?

Sponsoring is optional. If you are making money from the direct use of `CefSharp` then I'd ask that you help [Sponsor](https://github.com/sponsors/amaitland) the project.

**Reply by n303ef** [reply]
I would like to sponsor monthly, but unfortunately github payments are not working in our country
